ext/tk/extconf.rb uses rpath by default. Disable that.
--- a/ext/tk/extconf.rb
+++ b/ext/tk/extconf.rb
@@ -612,7 +612,7 @@ def libcheck_for_tclConfig(tcldir, tkdir
           $INCFLAGS << " -I" << File.join(File.dirname(File.dirname(file)),"include") if is_win32?
         else
           tcllibs = append_library($libs, libname)
-          tcllibs = "-L#{tcldir.quote} -Wl,-R#{tcldir.quote} " + tcllibs
+          tcllibs = "-L#{tcldir.quote} " + tcllibs
 
           # FIX ME: avoid pathname trouble (fail to find) on MinGW.
           $INCFLAGS << " -I" << File.join(File.dirname(tcldir),"include") if is_win32?
@@ -654,7 +654,7 @@ def libcheck_for_tclConfig(tcldir, tkdir
         else
           tklibs = append_library("", libname)
           #tklibs = append_library("", $1)
-          tklibs = "-L#{tkdir.quote} -Wl,-R#{tkdir.quote} " + tklibs
+          tklibs = "-L#{tkdir.quote} " + tklibs
 
           # FIX ME: avoid pathname trouble (fail to find) on MinGW.
           $INCFLAGS << " -I" << File.join(File.dirname(tcldir),"include") if is_win32?
@@ -1123,7 +1123,7 @@ def find_tcl(tcllib, stubs, version, *op
                   tcllibs = libs_param + " -DSTATIC_BUILD " + fname.quote
                 else
                   tcllibs = append_library($libs, lib_w_sufx)
-                  tcllibs = "-L#{path.quote} -Wl,-R#{path.quote} " + tcllibs
+                  tcllibs = "-L#{path.quote} " + tcllibs
                 end
                 if try_func(func, tcllibs, ["tcl.h"])
                   return [true, path, nil, tcllibs, *inc]
@@ -1262,7 +1262,7 @@ def find_tk(tklib, stubs, version, *opt_
                   tklibs = libs_param + " -DSTATIC_BUILD " + fname.quote
                 else
                   tklibs = append_library($libs, lib_w_sufx)
-                  tklibs = "-L#{path.quote} -Wl,-R#{path.quote} " + tklibs
+                  tklibs = "-L#{path.quote} " + tklibs
                 end
                 if try_func(func, tklibs, ["tcl.h", "tk.h"])
                   return [true, path, nil, tklibs, *inc]
@@ -1945,7 +1945,7 @@ $defs += collect_tcltk_defs(TclConfig_In
 # MacOS X Frameworks?
 if TkLib_Config["tcltk-framework"]
   puts("Use MacOS X Frameworks.")
-  ($LDFLAGS ||= "") << " -L#{TkLib_Config["tcl-build-dir"].quote} -Wl,-R#{TkLib_Config["tcl-build-dir"].quote}" if TkLib_Config["tcl-build-dir"]
+  ($LDFLAGS ||= "") << " -L#{TkLib_Config["tcl-build-dir"].quote}" if TkLib_Config["tcl-build-dir"]
 
   if tcl_cfg_dir
     TclConfig_Info['TCL_LIBS'] ||= ""
@@ -1970,7 +1970,7 @@ if TkLib_Config["tcltk-framework"]
     end
   end
 
-  $LDFLAGS  << " -L#{TkLib_Config["tk-build-dir"].quote} -Wl,-R#{TkLib_Config["tk-build-dir"].quote}" if TkLib_Config["tk-build-dir"]
+  $LDFLAGS  << " -L#{TkLib_Config["tk-build-dir"].quote}" if TkLib_Config["tk-build-dir"]
 
   if tk_cfg_dir
     TkConfig_Info['TK_LIBS'] ||= ""
